src/mlpack/methods/ann/layer/padding.hpp

```diff
@@ -42,19 +42,19 @@ class PaddingType : public Layer<MatType>
               const size_t padHBottom = 0);
 
   //! Clone the PaddingType object. This handles polymorphism correctly.
-  PaddingType* Clone() const { return new PaddingType(*this); }
+  PaddingType<MatType>* Clone() const { return new PaddingType<MatType>(*this); }
```

Comment:
I'm not sure these changes are necessary.  Did you have a compilation problem without them?  My understanding is that within the scope of the class definition, template arguments aren't strictly necessary; so, e.g., `PaddingType` should be equivalent to `PaddingType<MatType>`.

(Relevant legalese is section 14.6.1 of the [standard](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf); although I've linked to an old version there, I doubt it has changed much.)

Reply:
Yeh, guess you are right.  Not sure why I felt the need to make those changes; there were other places where the missing template args were necessary.  I will revert these back.